src/promptflow/promptflow/_sdk/_submitter/test_submitter.py

```diff
@@ -448,21 +496,91 @@ def exec_with_inputs(self, inputs):
             client=self._client,
         )
         storage = DefaultRunStorage(base_dir=self.flow.code, sub_dir=Path(".promptflow/intermediate"))
-        flow_executor = CSharpExecutorProxy.create(
-            flow_file=self.flow.path,
-            working_dir=self.flow.code,
+        flow_executor = async_run_allowing_running_loop(
+            CSharpExecutorProxy.create,
+            self.flow.path,
+            self.flow.code,
             connections=connections,
             storage=storage,
         )
 
         try:
             # validate inputs
             flow_inputs, _ = self.resolve_data(inputs=inputs, dataplane_flow=self.dataplane_flow)
-            line_result = async_run_allowing_running_loop(flow_executor.exec_line_async, inputs, index=0)
-            # line_result = flow_executor.exec_line(inputs, index=0)
-            if isinstance(line_result.output, dict):
+            line_result = async_run_allowing_running_loop(flow_executor.exec_line_async, inputs, index=0,
+                                                          allow_generator_output=allow_generator_output)
+            if not isinstance(line_result, AsyncGeneratorType) and isinstance(line_result.output, dict):
                 # Remove line_number from output
                 line_result.output.pop(LINE_NUMBER_KEY, None)
             return line_result
         finally:
             flow_executor.destroy()
+
+    def _chat_flow(self, inputs, chat_history_name, environment_variables: dict = None, show_step_output=False):
```

Comment:
what's the difference between _chat_flow for TestSubmitterViaProxy & TestSubmitter?

Reply:
1. connections resolved
2. move those to flow_test fucntion, since csharp need different handling for executor returned result 
![image](https://github.com/microsoft/promptflow/assets/26239730/98656934-cc84-4bc0-9fa6-09257ffc5b31)

3. no need resovle generator anymore since node_run_infos.output and run_info.output returned by excutor won't be generator, it's a dict like
https://github.com/microsoft/promptflow/pull/1837#discussion_r1464680162